pkg/kubelet/certificate/certificate_manager.go

```diff
@@ -160,9 +154,19 @@ func (m *manager) shouldRotate() bool {
 	m.certAccessLock.RLock()
 	defer m.certAccessLock.RUnlock()
 	notAfter := m.cert.Leaf.NotAfter
-	total := notAfter.Sub(m.cert.Leaf.NotBefore)
-	remaining := notAfter.Sub(time.Now())
-	return remaining < 0 || uint(remaining*100/total) < m.shouldRotatePercent
+	totalDuration := float64(notAfter.Sub(m.cert.Leaf.NotBefore))
+
+	// Use some jitter to set the rotation threshold so each node will rotate
+	// at approximately 70-90% of the total lifetime of the certificate.  With
+	// jitter, if a number of nodes are added to a cluster at approximately the
+	// same time (such as cluster creation time), they won't all try to rotate
+	// certificates at the same time for the rest of the life of the cluster.
+	//
+	// For the parameters of the Jitter function:
+	//    [total*0.7, total+total*0.7*0.28] ~ [total*0.7, total*0.9]
+	jitteryDuration := wait.Jitter(time.Duration(totalDuration*0.7), 0.28)
+	rotationThreshold := m.cert.Leaf.NotBefore.Add(jitteryDuration)
```

Comment:
About once an hour. Rather than saving the threshold I intend to modify the loop in a follow on PR so that it doesn't check periodically but instead just pauses for the correct duration.